client/src/client.rs

```diff
@@ -439,6 +439,10 @@ pub trait RpcApi: Sized {
         self.call("getblockhash", &[height.into()])
     }
 
+    fn get_block_stats(&self, height: u64) -> Result<json::GetBlockStatsResult> {
```

Comment:
This can also accept the block hash instead of the height, maybe worth supporting it?

There's also an option for selecting which fields you want statistics for. Though this would require changing all the `GetBlockStatsResult` fields to be an `Option<T>`, which isn't great.

Reply:
All of them?

Reply:
If we want to support getting a specific list of fields from `getblockstats`, then I'm afraid yes...

Perhaps it could be a separate struct? So we have `get_block_stats(height) -> GetBlockStatsResult` and `get_block_stats_fields(height, fields) -> GetBlockStatsResultPartial`.

Reply:
Okay, I proposed some code for this :-) Please check out my last commit @shesek

Reply:
Looks good! Commented with two small nits